cxx-qt-gen/test_outputs/invokables.h

```diff
@@ -36,13 +43,14 @@ class MyObject : public QObject
 
 private:
   rust::Box<MyObjectRust> m_rustObj;
-  std::mutex m_rustObjMutex;
+  std::shared_ptr<std::mutex> m_rustObjMutex;
   bool m_initialised = false;
+  std::shared_ptr<QPointer<MyObject>> m_cxxQtThreadPtr;
```

Comment:
So the point of a QPointer is that then the QObject is destroyed it doesn't become a dangling pointer like a normal raw pointer, otherwise it's exactly the same as a normal C++ pointer.

And for the second point, i don't think it'll try to call the QObject deconstructor? ... "Destroys the guarded pointer. Just like a normal pointer, destroying a guarded pointer does not destroy the object being pointed to." https://doc.qt.io/qt-6/qpointer.html#dtor.QPointer

But the question is probably do we even need this protection anyway? or is a raw pointer OK ... i'll check where we use things again...

Reply:
Changed to a raw pointer as there is no benefit for our usecase.